test/test_sparse.py

```diff
@@ -2363,6 +2363,304 @@ def test_sparse_to_numpy(self):
         t = torch.sparse_coo_tensor(torch.tensor(([0, 0], [2, 0])), torch.tensor([1, 4]))
         self.assertRaises(TypeError, lambda: t.numpy())
 
+    def test_softmax(self):
+        if self.device != 'cpu':
+            self.skipTest('sparse softmax not available on %s' % (self.device))
+
+        import torch.nn.functional as F
+
+        def to_dense(sparse, fill_value=None):
+            """
+            Return dense tensor from a sparse tensor using given fill value.
+            """
+            if isinstance(fill_value, str):
+                fill_value = float(fill_value)
+            if fill_value is None or fill_value == 0:
+                return sparse.to_dense()
+            sparse = sparse.coalesce()
+            dense = torch.empty(sparse.shape, dtype=sparse.dtype, device=sparse.device)
+            dense[:] = fill_value
+            for idx, value in zip(sparse._indices().t(), sparse._values()):
+                dense[tuple(idx)] = value
```

Comment:
This is going to be soooo sloooow. Probably OK for testing, but I bet you can probably do this faster by scattering zeros to the locations of `dense` referenced by `indices`, and then doing a dense-sparse addition. NO ACTION NECESSARY ON THIS COMMENT.